feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go

```diff
@@ -257,30 +320,53 @@ func testBackupDecap(ctx context.Context, t *testing.T, args *testArgs) {
 	if want := oc.Interface_OperStatus_DOWN; operStatus != want {
 		t.Errorf("Get(DUT port2 oper status): got %v, want %v", operStatus, want)
 	}
-
-	t.Log("Validate traffic passes through port3")
-	validateTrafficFlows(t, args.ate, backupFlow, baselineFlow)
+	t.Run("ValidateDecapPath", func(t *testing.T) {
+		t.Log("Validate traffic passes through ATE port4 after decap")
+		validateTrafficFlows(t, args.ate, []*ondatra.Flow{decapFlow}, []*ondatra.Flow{baseFlow}, decapFlowFliter)
+	})
 }
 
 // TE11.3 - case 2: new tunnel viability triggers decap in the backup NHG.
 func testDecapEncap(ctx context.Context, t *testing.T, args *testArgs) {
 
-	t.Logf("Adding NH %d with atePort2 via gRIBI", NH3ID)
-	args.client.AddNH(t, NH3ID, atePort2.IPv4, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
-	t.Logf("Adding NHG %d via gRIBI", NH3ID)
-	args.client.AddNHG(t, NH3ID, map[uint64]uint64{NH3ID: 100}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
-	t.Logf("Adding an IPv4Entry for %s pointing to atePort2 via gRIBI", secondaryTunnelDstIP)
-	args.client.AddIPv4(t, secondaryTunnelDstIP+"/"+mask, NH3ID, vrfName, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	t.Logf("Adding VIP %v/32 with NHG %d NH %d and  atePort2 via gRIBI", VIP1, NHG1ID, NH1ID)
+	args.client.AddNH(t, NH1ID, atePort2.IPv4, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	args.client.AddNHG(t, NHG1ID, map[uint64]uint64{NH1ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	args.client.AddIPv4(t, VIP1+"/"+mask, NHG1ID, *deviations.DefaultNetworkInstance, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+
+	t.Logf("Adding VIP %v/32 with NHG %d , NH %d and  atePort3 via gRIBI", VIP2, NHG2ID, NH2ID)
+	args.client.AddNH(t, NH2ID, atePort3.IPv4, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	args.client.AddNHG(t, NHG2ID, map[uint64]uint64{NH2ID: 1}, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+	args.client.AddIPv4(t, VIP2+"/"+mask, NHG2ID, *deviations.DefaultNetworkInstance, *deviations.DefaultNetworkInstance, fluent.InstalledInFIB)
+
+	t.Logf("Adding NHG %d with NH %d as decap and DEFAULT vrf lookup via gRIBI", NHG100ID, NH100ID)
```

Comment:
Typo in the log message? NHG100 does not do decap.

Reply:
corrected